src/software/ai/evaluation/scoring/skills/skill_graph.cpp

```diff
@@ -0,0 +1,109 @@
+#include "software/ai/evaluation/scoring/skills/skill_graph.h"
+
+#include "software/util/generic_factory/generic_factory.h"
+
+SkillGraph::SkillGraph(std::shared_ptr<Strategy> strategy)
```

Comment:
iiirc, we will only have a finite number of "Skill" instances over the lifetime of the game?

Reply:
Yeah, SkillGraph creates one of each Skill using the GenericFactory. We get a shared ptr to the next Skill to execute from the SkillGraph. We request primitives for a given robot from the Skill; the Skill will keep track of an FSM for that robot and use that FSM to generate primitives for that robot.